src/ripple/net/impl/RPCCall.cpp

```diff
@@ -144,6 +145,21 @@ class RPCParser
         }
     }
 
+    static Json::Value validPublicKey (std::string const& strPk)
+    {
+        if (parseBase58<PublicKey> (TokenType::AccountPublic, strPk))
+            return true;
+
+        auto pkHex = strUnHex (strPk);
+        if (!pkHex)
+            return false;
+
+        if (!publicKeyType(makeSlice(*pkHex)))
+            return false;
+
+        return true;
```

Comment:
An alternative here is this:

```
if (auto k = strUnHex(strPk))
    return static_cast<bool>(publicKeyType(makeSlice(*k)));
return false;
```

Note that this rewrite returns `true` from the branch that _positively_ identifies a key as being of a given type, instead of by default. I think that's a net positive. Apart from that, I don't know if it's better; it is _shorter_, but that isn't always a positive.

Reply:
@nbougalis do you mind if we leave this as is for this patch? The reason I suggest this is that this method is simply extracted verbatim from the **parseFetchInfo** method below. This way we can just focus on extracting the common logic here, and refactoring / cleaning this up in a separate PR.

Reply:
Not at all. Whether review comments should be "resolved" is typically up to the PR author. Obviously, exceptions for bugs/problems exist, but this wasn't that. I'm fine with leaving this as is.